tools/pubsys/src/aws/ssm/mod.rs

```diff
@@ -156,8 +210,56 @@ pub(crate) async fn run(args: &Args, ssm_args: &SsmArgs) -> Result<()> {
     Ok(())
 }
 
+// Determines the number of AMIs that can be checked for public permissions in parallel.
+const CONCURRENT_AMI_CHECKS: usize = 16;
```

Comment:
Would it be better to set this at runtime based on the running host's number of cores?

Reply:
Since this is network-bound, I picked an arbitrarily "low" number of concurrent checks. I've found that you often hit throttle limits trying to hit AWS using all of your cores if your host is too beefy. This can be rectified with more careful throttling avoidance, but I opted for a simpler route here.

I should probably clarify why the number was chosen in a comment.

Reply:
Interesting, I was actually worried this number was too high. I was thinking something like `Min(cpu_cores, 16)`, but even wondering if that should be more like `8` than `16`.

Reply:
`8` is probably a safer bet!